llvm/lib/Transforms/Utils/LoopUtils.cpp

```diff
@@ -1930,7 +1930,15 @@ llvm::hasPartialIVCondition(const Loop &L, unsigned MSSAThreshold,
   if (!TI || !TI->isConditional())
     return {};
 
-  auto *CondI = dyn_cast<CmpInst>(TI->getCondition());
+  Instruction *CondI = nullptr;
```

Comment:
I think you can write this more simply as:

```
Instruction *CondI = dyn_cast<CmpInst>(TI->getCondition());
```

Reply:
Done